mongo_orchestration/replica_sets.py

```diff
@@ -59,34 +59,61 @@ def __init__(self, rs_params):
         if not not self.sslParams:
             self.kwargs['ssl'] = True
 
-        config = {"_id": self.repl_id, "members": [
-                  self.member_create(member, index) for index, member in enumerate(rs_params.get('members', {}))
-                  ]}
+        # Always use a keyFile if auth is on.
+        if self.login and not self.auth_key:
+            self.auth_key = DEFAULT_AUTH_KEY
+
+        # Find a server that is suitable to initiate the set.
+        other_members = []
+        members = rs_params.get('members', [])
+        for index, member in enumerate(rs_params.get('members', [])):
+            if not (member.get('arbiterOnly', False)
```

Comment:
Are you sure this is necessary? The set is uninitialized, so it may not matter that the host you're running initiate on can't ultimately be a primary.